cratedb_toolkit/io/kinesis/adapter.py

```diff
@@ -0,0 +1,56 @@
+import asyncio
+import typing as t
+
+from aiobotocore.session import AioSession
+from kinesis import Consumer, JsonProcessor, Producer
+from yarl import URL
+
+
+class KinesisAdapter:
+    def __init__(self, kinesis_url: URL):
+        self.session = AioSession()
+        self.session.set_credentials(access_key=kinesis_url.user, secret_key=kinesis_url.password)
+        self.endpoint_url = None
+        if kinesis_url.host and kinesis_url.host.lower() != "aws":
+            self.endpoint_url = f"http://{kinesis_url.host}:{kinesis_url.port}"
+        self.kinesis_url = kinesis_url
+        self.region_name = kinesis_url.query.get("region")
+        self.stream_name = self.kinesis_url.path.lstrip("/")
+
+    def consume_forever(self, handler: t.Callable):
+        asyncio.run(self._consume_forever(handler))
+
+    async def _consume_forever(self, handler: t.Callable):
+        """
+        Consume items from a Kinesis stream.
+        """
+        async with Consumer(
+            stream_name=self.stream_name,
+            session=self.session,
+            endpoint_url=self.endpoint_url,
+            region_name=self.region_name,
+            # TODO: Make configurable.
+            create_stream=True,
+            iterator_type="TRIM_HORIZON",
+            sleep_time_no_records=0.2,
```

Comment:
Yes. What the comment says.

Reply:
`iterator_type="TRIM_HORIZON"` (currently hard-coded) means it will always read the Stream from its starting point. Sure enough, this is probably the most important detail to be made configurable.

Reply:
07b19749f adds a few options to configure details when connecting to the Kinesis Stream.